cmd/openfga/main_test.go

```diff
@@ -294,6 +322,91 @@ func connect(t *testing.T, tester OpenFGATester) *grpc.ClientConn {
 	return conn
 }
 
+func MigrateDatastoreToVersion4Test(t *testing.T, engine string) {
+	t.Run(engine, func(t *testing.T) {
+		store := ulid.Make().String()
+		t.Logf("start the database container and run migrations up to the latest migration version available")
+		testDatastore := storagefixtures.RunDatastoreTestContainer(t, engine)
+		uri := testDatastore.GetConnectionURI(true)
+
+		t.Logf("migrate database down to version 3")
+		migrateCommand := cmd.NewMigrateCommand()
+		migrateCommand.SetArgs([]string{"--datastore-engine", engine, "--datastore-uri", uri, "--version", strconv.Itoa(3)})
+		err := migrateCommand.Execute()
+		require.NoError(t, err)
+
+		t.Logf("start openfga 1.1.0 and wait for it to be ready")
+		// this is a hack to have the networking work in Github Actions
+		datastoreConnectionURI := strings.Replace(uri, "localhost", defaultIpAddressInLinux, -1)
+		tester, err := newOpenFGATester(t, "openfga/openfga:v1.1.0", "--datastore-engine", engine, "--datastore-uri", datastoreConnectionURI)
+		require.NoError(t, err)
+		defer tester.Cleanup()
+
+		t.Logf("create a client of openfga")
+		conn := connect(t, tester)
+		defer conn.Close()
+
+		client := openfgapb.NewOpenFGAServiceClient(conn)
+
+		t.Logf("write authorization model")
+		_, err = client.WriteAuthorizationModel(context.Background(), &openfgapb.WriteAuthorizationModelRequest{
+			StoreId: store,
+			TypeDefinitions: []*openfgapb.TypeDefinition{
+				{
```

Comment:
Please use the parser with a DSL string instead. I've been trying to gradually move things that way in PRs.